src/Writer.zig

```diff
@@ -183,6 +183,17 @@ pub fn text(writer: *Writer, s: []const u8) anyerror!void {
     writer.state = .text;
 }
 
+// insert some existing XML document without escaping anything
+pub fn embed(writer: *Writer, s: []const u8) anyerror!void {
+    switch (writer.state) {
+        .after_structure_end, .text => {},
+        .element_start => try writer.raw(">"),
+        .start, .after_bom, .after_xml_declaration, .end => unreachable,
+    }
+    try writer.raw(s);
+    writer.state = .after_structure_end;
```

Comment:
```suggestion
    writer.state = switch (writer.state) {
        .start, .after_bom, .after_xml_declaration => .after_xml_declaration,
        .element_start, .after_structure_end, .text => .text,
        .end => .end,
    };
```
Once this function is opened up to work in more scenarios, other possible states also need to be handled. In this suggestion, I chose `text` as the ending state within the root element to avoid indentation if another element is started right after this function (to maximize the possible uses, including for embedding raw text nodes).